include/mptcpd/plugin.h

```diff
@@ -19,17 +19,98 @@
 extern "C" {
 #endif
 
+struct sockaddr;
+struct mptcpd_pm;
+struct mptcpd_interface;
+
 /**
- * @brief Mptcpd plugin @a "symbol" argument for @c L_PLUGIN_DEFINE.
+ * @brief Symbol name of mptcpd plugin characterstics.
  *
- * All mptcpd plugins are expected to use this value as the @a symbol
- * argument in their @c L_PLUGIN_DEFINE() macro call.
+ * @note This is a private preprocessor constant that is not part of
+ *       the mptcpd plugin API.
  */
-#define MPTCPD_PLUGIN_DESC mptcpd_plugin_desc
+#define MPTCPD_PLUGIN_SYM _mptcpd_plugin
 
-struct sockaddr;
-struct mptcpd_pm;
-struct mptcpd_interface;
+/**
+ * @brief Define mptcpd plugin characterstics.
+ *
+ * Mptcpd plugins should use this macro to define and export
+ * characterstics (descriptor) required by mptcpd.
+ *
+ * @param[in] name        Plugin name (unquoted)
+ * @param[in] description Plugin description
+ * @param[in] priority    Plugin priority, where the higher values are
+ *                        lower in priority, and lower values are
+ *                        higher in priority, similar to how process
+ *                        scheduling priorities are defined.  Mptcpd
+ *                        defines convenience plugin priorities
+ *                        @c MPTCP_PLUGIN_PRIORITY_LOW,
+ *                        @c MPTCP_PLUGIN_PRIORITY_DEFAULT, and
+ *                        @c  MPTCP_PLUGIN_PRIORITY_HIGH.
+ * @param[in] init        Function called when mptcpd initializes the
+ *                        plugin.
+ * @param[in] exit        Function called when mptcpd finalizes the
+ *                        plugin.
+ */
+#define MPTCPD_PLUGIN_DEFINE(name, description, priority, init, exit)   \
+        extern struct mptcpd_plugin_desc const MPTCPD_PLUGIN_SYM        \
+                __attribute__((visibility("default")));                 \
+        struct mptcpd_plugin_desc const MPTCPD_PLUGIN_SYM = {           \
+                #name,                                                  \
+                description,                                            \
+                0, /* version */                                        \
+                priority,                                               \
+                init,                                                   \
+                exit                                                    \
+        };
+
+/// Lowest priority value, i.e. highest priority
+#define MPTCPD_PLUGIN_PRIORITY_LOW     -20
```

Comment:
The low/high inversion is based on the same inversion in the former ELL plugin framework.  Both are similar to the CPU scheduling priority values where higher priority has a lower numerical value.  In any case, I'm fine with making lower priority have a lower numerical value, and similarly for higher priorities, and will make the change.

Reply:
The inversion is fine - I was just saying to make MPTCP_PLUGIN_PRIORITY_LOW be 19 and _HIGH be -20, so that the human meaning matched the defined symbol.

Line 68 above is defining MPTCPD_PLUGIN_PRIORITY_LOW to be a high priority numeric value, which is confusing to human readers.

Reply:
Commits 3ed998eb21fa3d337ea98406812f95886f52bb5f and 110a0a821bf397341434681a5da263757dd803c5 implement your suggested change.